iFlight Blitz F745 AIO (#23985)
ArduPilot/ardupilot: libraries/AP_HAL_ChibiOS/hwdef/BlitzF7/README.md

# iFlight BLITZ Whoop F7 AIO Flight Controller

https://shop.iflight-rc.com/BLITZ-Whoop-F7-AIO-Pro1927

> The Beast H7 AIO is a flight controller produced by [iFlight](https://shop.iflight-rc.com/).

Reviewer: ?????

Reviewer 2: fixed

## Features

 - MCU: BGA-STM32F745, 216MHz
 - Gyro: DPS310
 - 16Mb Onboard Flash
 - BEC output: 5V 2.5A
 - Barometer: DPS310 or None
 - OSD: AT7456E
 - 5 UARTS: (UART1, UART2, UART3, UART4, UART7)
 - I2C for external compass. UART3 pins are used for I2C (BRD_ALT_CONFIG=1)
 - 5 PWM outputs (4 motors and 1 LED)

## Pinout

> ![BLITZ Whoop F7 AIO Board](blitz_f7_pinou.jpg "BLITZ Whoop F7 AIO")

Reviewer: really need something that either lists all pins or a diagram with all pad labels shown

# iFlight BLITZ Whoop F7 AIO Flight Controller

https://shop.iflight-rc.com/BLITZ-Whoop-F7-AIO-Pro1927

The Blitz F7 AIO is a flight controller produced by [iFlight](https://shop.iflight-rc.com/).

## Features

> 

Reviewer: ```suggestion

  - Integrated 55A 4 in 1 ESC
```

Reviewer 2: Fixed

 - MCU: BGA-STM32F745, 216MHz
 - Gyro: DPS310
 - 16Mb Onboard Flash
 - BEC output: 5V 2.5A
 - Barometer: DPS310 or None
 - OSD: AT7456E
 - 5 UARTS: (UART1, UART2, UART3, UART4, UART7)
 - I2C for external compass.

>  - 5 PWM outputs (4 motors and 1 LED)

Reviewer: ```suggestion
 - 5 PWM outputs (4 used internally for integrated 4in1 ESC and 1 LED)
```

## Pinout

![BLITZ Whoop F7 AIO Board](blitz_f7_pinout.jpg "BLITZ Whoop F7 AIO")

## UART Mapping

The UARTs are marked Rn and Tn in the above pinouts. The Rn pin is the
receive pin for UARTn. The Tn pin is the transmit pin for UARTn.
|Name|Pin|Function|
|:-|:-|:-|
|SERIAL0|COMPUTER|USB|
|SERIAL1|RX1/TX1|UART1 (DJI connector, DMA-enabled)|
|SERIAL2|TX2/RX2|UART2 (DJI connector RX, DMA-enabled, TX is on the back side of board)|
|SERIAL3|TX3/RX3|UART3 (Alt RX, DMA-enabled)|
|SERIAL4|TX4/RX4|UART4|

> |SERIAL7|TX7/RX7|UART7 (GPS)|

Reviewer: this will not work by default...you would need to change protocol on Serial 3 and 4 and set the default for Serial 7 in hwdef to GPS protocol

Reviewer 2: Thanks fixed.

## RC Input

RC input is configured on the (UART2_RX/UART2_TX) pins which forms part of the DJI connector. It supports all RC protocols.

## OSD Support

The BLITZ Whoop F7 AIO supports OSD using OSD_TYPE 1 (MAX7456 driver).

## PWM Output

> The BLITZ Whoop F7 AIO supports up to 4 PWM outputs. The pads for motor output ESC1 to ESC4 on the above diagram are for the 4 outputs. All 4 outputs support DShot as well as all PWM types.

Reviewer: ```suggestion
The BLITZ Whoop F7 AIO has 4 PWM outputs internally connected to its 4 in 1 ESC. The pads for motor output are M1 to M4 on the board . All 4 outputs support BiDirDShot and DShot, as well as all PWM types. The default configuration is for BiDirShot using the already installed BlueJay firmware.
```

Reviewer 2: Fixed

The PWM are in in two groups:

 - PWM 1-4 in group1
 - PWM 5   in group2

Channels within the same group need to use the same output rate. If
any channel in a group uses DShot then all channels in the group need
to use DShot.

## Battery Monitoring

> The board has a builtin voltage sensor. The voltage sensor can handle up to 6S

Reviewer: ```suggestion
The board has a builtin voltage sensor and a current sensor input tied to its 4 in 1 ESC current sensor. The voltage sensor can handle up to 6S.
```

Reviewer 2: Fixed

LiPo batteries.

The correct battery setting parameters are:

 - BATT_MONITOR 4
 - BATT_VOLT_PIN 12

>  - BATT_VOLT_MULT around 10.9

Reviewer: ```suggestion
 - BATT_VOLT_MULT  10.9
```

Reviewer 2: Fixed

 - BATT_CURR_PIN 13

>  - BATT_CURR_MULT around 50

Reviewer: ```suggestion
 - BATT_CURR_MULT  50
```

Reviewer 2: Fixed

These are set by default in the firmware and shouldn't need to be adjusted

## Compass

> The BLITZ Whoop F7 AIO does not have a builtin compass, but you can attach an external compass to I2C pins.

Reviewer: I see no pads for I2C...if true, then compass is not possible...and there should be a caveat about its restricted capabilities

Reviewer 2: I2C pads are on the corner - just look closely

## NeoPixel LED

The board includes a NeoPixel LED on the underside which is pre-configured to output ArduPilot sequences. This is the fifth PWM output.

> Currently this output is not functional.

Reviewer: looks to me to be defined as functional and setup by default for NEOPIXEL

Reviewer 2: Removed
